lib/matplotlib/legend.py

```diff
@@ -447,20 +443,17 @@ def __init__(
         #: instance.
         self._custom_handler_map = handler_map
 
-        def val_or_rc(val, rc_name):
-            return val if val is not None else mpl.rcParams[rc_name]
```

Comment:
I think you could possibly save a bunch of method gets below by setting this and reusing it?
`val_or_rc = mpl._val_or_rc`

Reply:
I would like to warn on ineffective micro optimizations. Squeezing out nanoseconds does not make any measureable difference.

A function lookup cost 30ns (Python 3.11):
```
In [2]: %timeit mpl.rc_params_from_file
29.7 ns ± 0.544 ns per loop
```

If we want to improve performance (and there's a lot to gain) that would need to be guided by profiling to identify the real pain points. In other cases, code structure/readability should be the primary guideline. - Not shure what is better here, but I tend to not do the additional logic indirection for saving a method to a new variable just to save method lookups.